apps/provider-console/src/components/become-provider/WalletImport.tsx

```diff
@@ -0,0 +1,334 @@
+"use client";
+import React, { useEffect, useState } from "react";
+import { useForm } from "react-hook-form";
+import {
+  Button,
+  Form,
+  FormControl,
+  FormDescription,
+  FormField,
+  FormItem,
+  FormLabel,
+  FormMessage,
+  RadioGroup,
+  RadioGroupItem,
+  Separator,
+  Textarea
+} from "@akashnetwork/ui/components";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { useAtom } from "jotai";
+import { HomeIcon } from "lucide-react";
+import { useRouter } from "next/router";
+import { z } from "zod";
+
+import providerProcessStore from "@src/store/providerProcessStore";
+import restClient from "@src/utils/restClient";
+import { ResetProviderForm } from "./ResetProviderProcess";
+
+function decodeBase64(base64: string): string {
+  return Buffer.from(base64, "base64").toString("utf-8");
+}
+
+async function encrypt(data: string, publicKey: string): Promise<string> {
+  const { default: JSEncrypt } = await import("jsencrypt");
+  const encryptor = new JSEncrypt();
+
+  const decodedPublicKey = decodeBase64(publicKey);
+  encryptor.setPublicKey(decodedPublicKey);
+
+  const encryptedData = encryptor.encrypt(data);
+  if (!encryptedData) {
+    throw new Error("Encryption failed");
+  }
+  return encryptedData;
+}
+
+interface WalletImportProps {
+  stepChange: () => void;
+}
+
+const appearanceFormSchema = z.object({
+  walletMode: z.enum(["seed", "manual"], {
+    required_error: "Please select a mode."
+  })
+});
+
+const seedFormSchema = z.object({
+  seedPhrase: z.string().refine(
+    value => {
+      const wordCount = value.trim().split(/\s+/).length;
+      return wordCount === 12 || wordCount === 24;
+    },
+    {
+      message: "Seed phrase must be either 12 or 24 words."
+    }
+  )
+});
+
+type AppearanceFormValues = z.infer<typeof appearanceFormSchema>;
+type SeedFormValues = z.infer<typeof seedFormSchema>;
+
+export const WalletImport: React.FC<WalletImportProps> = () => {
+  const [mode, setMode] = useState<string>("");
+  const [isMounted, setIsMounted] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState<string | null>(null);
+  const router = useRouter();
+
+  const [providerProcess] = useAtom(providerProcessStore.providerProcessAtom);
+  const [, resetProviderProcess] = useAtom(providerProcessStore.resetProviderProcess);
+
+  const defaultValues: Partial<AppearanceFormValues> = {
+    walletMode: "seed"
+  };
+
+  const form = useForm<AppearanceFormValues>({
+    defaultValues
+  });
+
+  const seedForm = useForm<SeedFormValues>({
+    resolver: zodResolver(seedFormSchema)
+  });
+
+  const submitData = async (data: AppearanceFormValues) => {
+    setMode(data.walletMode);
+  };
+
+  const submitForm = async (data: SeedFormValues) => {
+    setIsLoading(true);
+    setError(null);
+    try {
+      if (providerProcess.machines && providerProcess.machines.length > 0) {
```

Comment:
the code would look a bit simpler if fn stopped early on the inverted condition here

Reply:
Refactored the code